firedrake/__init__.py

```diff
@@ -2,18 +2,20 @@
 import os
 import sys
 config = firedrake_configuration.get_config()
-if "PETSC_DIR" in os.environ and not config["options"]["honour_petsc_dir"]:
-    if os.environ["PETSC_DIR"] != os.path.join(sys.prefix, "src", "petsc")\
-       or os.environ["PETSC_ARCH"] != "default":
-        raise ImportError("PETSC_DIR is set, but you did not install with --honour-petsc-dir.\n"
-                          "Please unset PETSC_DIR (and PETSC_ARCH) before using Firedrake.")
-elif "PETSC_DIR" not in os.environ and config["options"]["honour_petsc_dir"]:
-    raise ImportError("Firedrake was installed with --honour-petsc-dir, but PETSC_DIR is not set.\n"
-                      "Please set PETSC_DIR (and PETSC_ARCH) before using Firedrake.")
-elif not config["options"]["honour_petsc_dir"]:  # Using our own PETSC.
-    os.environ["PETSC_DIR"] = os.path.join(sys.prefix, "src", "petsc")
-    os.environ["PETSC_ARCH"] = "default"
-del sys, config
+# This check is only valid for the default install method, not when using Spack
+if config["options"].get("install_mode", "default") == "default":
+    if "PETSC_DIR" in os.environ and not config["options"]["honour_petsc_dir"]:
+        if os.environ["PETSC_DIR"] != os.path.join(sys.prefix, "src", "petsc")\
+           or os.environ["PETSC_ARCH"] != "default":
+            raise ImportError("PETSC_DIR is set, but you did not install with --honour-petsc-dir.\n"
+                              "Please unset PETSC_DIR (and PETSC_ARCH) before using Firedrake.")
+    elif "PETSC_DIR" not in os.environ and config["options"]["honour_petsc_dir"]:
+        raise ImportError("Firedrake was installed with --honour-petsc-dir, but PETSC_DIR is not set.\n"
+                          "Please set PETSC_DIR (and PETSC_ARCH) before using Firedrake.")
+    elif not config["options"]["honour_petsc_dir"]:  # Using our own PETSC.
+        os.environ["PETSC_DIR"] = os.path.join(sys.prefix, "src", "petsc")
+        os.environ["PETSC_ARCH"] = "default"
+del os, sys, config
```

Comment:
I think we should not delete `os` here? I think we still need it in line 126 in `_omp_num_threads = int(os.environ.get('OMP_NUM_THREADS'))`? If if I am wrong you should take it out of the del in the last line of the file however.

Reply:
Great spot!